lib/truffle/truffle/patching.rb

```diff
@@ -0,0 +1,38 @@
+module Truffle::Patching
+  TRUFFLE_PATCHES_DIRECTORY = File.join(Truffle::Boot.ruby_home, 'lib', 'patches')
+  TRUFFLE_PATCHES           = Dir.glob(File.join(TRUFFLE_PATCHES_DIRECTORY, '**', '*.rb')).
+      select { |path| File.file? path }.
+      map { |path| path[(TRUFFLE_PATCHES_DIRECTORY.size+1)..-4] }.
+      reduce ({}) { |h, v| h.update v => true }
```

Comment:
`Array#to_h` might be a good compromise here, with `[path[...], true]` in the map block.